csrc/type.h

```diff
@@ -995,7 +995,7 @@ inline PolymorphicValue castToDtype(
 
 // Convert an enum to its underlying type.
 template <typename E>
-constexpr auto toUnderlying(E e) noexcept {
+constexpr auto castEnumToUnderlyingType(E e) noexcept {
```

Comment:
Just double check my understanding here.

I see that we are using `castEnumToUnderlyingType` to replace `serde::mapToSerdeDtype`, which has a few overloads on different dtypes.

I guess our assumption here is that, we'll always serialize the one object to a int and deserialize the int back to the same object. So even though the `long dtype` thing could carry different objects/types in the serialized form. The deserialized object would still be consistent so we are all good.

Reply:
> It is replaced with a cast to the NvFuser Enum's underlying type. During deserialization, we cast the underlying type back to the correct NvFuser enum.

So this is not just the NvFuser enum, since we are also handling the ScalarType from aten? Might want to update the PR description.